tests/util.go

```diff
@@ -33,44 +47,64 @@ func ProcessTemplateWithParameters(srcFilePath, dstFilePath string, params ...st
 	return filePath
 }
 
-func CreateResourceWithFilePathTestNamespace(resourceType, resourceName, filePath string) {
-	createResourceWithFilePath(resourceType, resourceName, filePath, NamespaceTestDefault)
+func CreateResourceWithFilePathTestNamespace(filePath string) {
+	exec(Result{verb: "create", filePath: filePath, nameSpace: NamespaceTestDefault})
 }
 
-func createResourceWithFilePath(resourceType, resourceName, filePath, nameSpace string) {
-	By(fmt.Sprintf("Creating %s:%s from the json file with the oc-create command", resourceType, resourceName))
-	_, err := ktests.RunOcCommand("create", "-f", filePath, "-n", nameSpace)
-	Expect(err).ToNot(HaveOccurred())
-	Eventually(func() bool {
-		out, err := ktests.RunOcCommand("get", resourceType, "-n", nameSpace)
-		Expect(err).ToNot(HaveOccurred())
-		return strings.Contains(out, resourceName)
-	}, time.Duration(2)*time.Minute).Should(BeTrue(), fmt.Sprint("Timed out waiting for %s to appear", resourceType))
+func DeleteResourceWithLabelTestNamespace(resourceType, resourceLabel string) {
+	By(fmt.Sprintf("Deleting %s:%s from the json file with the oc-delete command", resourceType, resourceLabel))
+	exec(Result{verb: "delete", resourceType: resourceType, resourceLabel: resourceLabel, nameSpace: NamespaceTestDefault})
 }
 
 func WaitUntilResourceReadyByNameTestNamespace(resourceType, resourceName, query, expectOut string) {
 	By(fmt.Sprintf("Wait until %s:%s ready", resourceType, resourceName))
-	waitUntilResourceReady(resourceType, "", resourceName, query, expectOut, NamespaceTestDefault)
+	exec(Result{verb: "get", resourceType: resourceType, resourceName: resourceName, query: query, expectOut: expectOut, nameSpace: NamespaceTestDefault})
 }
 
 func WaitUntilResourceReadyByLabelTestNamespace(resourceType, label, query, expectOut string) {
 	By(fmt.Sprintf("Wait until label=%s ready", label))
-	waitUntilResourceReady(resourceType, label, "", query, expectOut, NamespaceTestDefault)
+	exec(Result{verb: "get", resourceType: resourceType, resourceLabel: label, query: query, expectOut: expectOut, nameSpace: NamespaceTestDefault})
```

Comment:
Maybe the msg above (L65) would be clearer if the resource was added to it since many different cdi resources will have the same label?  Eg `By(fmt.Sprintf("Wait until resource %s with label=%s is ready", resourceType, label))`